src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -63,37 +65,65 @@ const getPlaceholderForType = (type: ActionType) => {
       return 'username or channel';
     case ActionType.PAGERDUTY:
       return 'service';
+    case ActionType.INTEGRATION:
+      return 'integration';
     default:
       throw Error('Not implemented');
   }
 };
+
 /**
- * Lists saved actions as well as control to add a new action
+ * When a new action is added, all of it's settings should be set to their default values.
+ * @param actionConfig
  */
-class ActionsPanel extends React.PureComponent<Props> {
-  /**
-   * Actions have a type (e.g. email, slack, etc), but only some have
-   * an integrationId (e.g. email is null). This helper creates a unique
-   * id based on the type and integrationId so that we know what action
-   * a user's saved action corresponds to.
-   */
-  getActionUniqueKey({type, integrationId}: Pick<Action, 'type' | 'integrationId'>) {
-    return `${type}-${integrationId}`;
-  }
+const getCleanAction = (actionConfig): Action => {
+  return {
+    type: actionConfig.type,
+    targetType:
+      actionConfig &&
+      actionConfig.allowedTargetTypes &&
+      actionConfig.allowedTargetTypes.length > 0
+        ? actionConfig.allowedTargetTypes[0]
+        : null,
+    targetIdentifier: '',
+    integrationId: actionConfig.integrationId,
+    inputType: actionConfig.inputType,
+    options: actionConfig.options || null,
+  };
+};
 
-  /**
-   * Creates a human-friendly display name for the integration based on type and
-   * server provided `integrationName`
-   *
-   * e.g. for slack we show that it is slack and the `integrationName` is the workspace name
-   */
-  getFullActionTitle({
-    type,
-    integrationName,
-  }: Pick<MetricActionTemplate, 'type' | 'integrationName'>) {
-    return `${ActionLabel[type]}${integrationName ? ` - ${integrationName}` : ''}`;
-  }
+/**
+ * Actions have a type (e.g. email, slack, etc), but only some have
+ * an integrationId (e.g. email is null). This helper creates a unique
+ * id based on the type and integrationId so that we know what action
+ * a user's saved action corresponds to.
+ */
+const getActionUniqueKey = ({
+  type,
+  integrationId,
+}: Pick<Action, 'type' | 'integrationId'>) => {
+  return `${type}-${integrationId}`;
+};
+
+// TODO MARCOS might use (internal/private) here
+/**
+ * Creates a human-friendly display name for the integration based on type and
+ * server provided `integrationName`
+ *
+ * e.g. for slack we show that it is slack and the `integrationName` is the workspace name
+ */
+const getFullActionTitle = ({
+  type,
+  integrationName,
+}: Pick<MetricActionTemplate, 'type' | 'integrationName'>) => {
+  return `${ActionLabel[type]}${integrationName ? ` - ${integrationName}` : ''}`;
```

Comment:
so I personally prefer something like:

![Screen Shot 2020-08-19 at 6 39 44 PM](https://user-images.githubusercontent.com/15368179/90708435-aa420000-e24e-11ea-9408-2238d80dc121.png)

which I just did by adding an `isInternal` field on the action response and then doing

```javascript
if (type == 'integration') {
    return `${integrationName}${isInternal ? ` (internal)` : ''}`;
  }
```